docs/acceptance-criteria/uacs-rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,503 @@
+  determine validity of the challenge.
+
+| Group creation - DKG challenge
+| When determining validity of the challenge, the following aspects should be
+  taken into account: +
+  - does it reference existing DKG result (if not, then challenge invalid) +
+  - within or outside of the challenge period for the specified DKG result (if
+  outside, then invalid) +
+  - have all members who signed a DKG result been selected by the sortition pool
+  to be in the group (if yes, then challenge invalid).
+
+| Group creation - DKG challenge
+| If the challenge notification was received within the challenge period and was
+  deemed justified, then: +
+  - the malicious DKG result should be immediately discarded (in the same
+  transaction in which notification happened) +
+  - all sortition pool members who signed the result should be slashed (fixed
+  amount) (in the same tx) +
+  - notifier should receive 5% from the total slashed amount (in the same tx) +
+  - remaning 95% of the total slashed amount shoud be burned +
+  - the members of the signing group should be given another chance to publish
+  the DKG result +
+  - DKG timeout timer and the result submission eligibility order should be
+  reset.
+
+| Group creation - DKG challenge
+| If the challenge notification was received within the challenge period and was
+  not justified, then: +
+  - challenge transaction is reverted.
+
+| Group creation - DKG challenge
+| If the challenge notification (justified or not) was received outside of the
+  challenge period, then: +
+  - challenge transaction is reverted.
+
+| Group creation - DKG acceptance
+| Anyone can send a transaction requesting unlocking of the sortition pool and
+  requesting marking of the DKG result as accepted.
+
+| Group creation - DKG acceptance
+| Upon processing the transaction unlocking the sortition pool and marking the
+  DKG result as accepted, the random beacon contract should determine validity
+  of the transaction.
+
+| Group creation - DKG acceptance
+| When determining validity of the tx unlocking the sortition pool and accepting
+  the DKG result, the following aspects should be taken into account: +
+  - is the sortition pool locked? +
+  - has the challenge period already passed? +
+  Only if all the above conditions have been met, the tx is considered valid.
+
+| Group creation - DKG acceptance
+| If the transaction unlocking the sortition pool and marking the DKG result as
+  accepted gets approved, then: +
+  - the sortition pool should get unlocked +
+  - the DKG result should be accepted and a group should be created based on the
+  candidate group +  
+  - submitter should receive the fixed reward (in T), paid from the maintenance
+  pool to the submitter’s address (in the same tx) +
+  - operators who became eligible for submitting the DKG result before the
+  member who submitted the DKG result should be punished by being excluded from
+  earning rewards for a fixed amount of time (in the same tx).
```

Comment:
There is one additional scenario here not covered by the RFC: we need to have a fixed timeout under which DKG result submitter is expected to call the approve transaction. If they do not do that, anyone becomes eligible to call the approve transaction. The one who called that transaction receives the submitter reward.

This protection is needed because DKG result approve transaction might not be that cheap to call and we want to have everyone incentivized to call it in case DKG result submitter decides not doing that.

Reply:
Clarified who can mark the result as accepted in commit 8c9a2ffd19dd2435957041357aaaabc3aa4d75cd. Should this fixed timeout be a governable param? If yes, I need to add a UAC for it.

Reply:
Atill not sure if that's governable or not. Asked also in https://github.com/keep-network/keep-core/pull/2751/files#r762768804.